sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileSystemsTest.java

```diff
@@ -186,6 +189,56 @@ public void testRenameIgnoreMissingFiles() throws Exception {
         containsInAnyOrder("content3"));
   }
 
+  @Test
+  public void testRenameOverwriteExistingFilesThrows() throws Exception {
+    // Simulate a filesystem which will not overwrite existing files
+    FileSystem<LocalResourceId> fs = new LocalFileSystem(new CopyOption[0]);
+    thrown.expect(FileAlreadyExistsException.class);
+    executeRename(fs, false);
+  }
+
+  @Test
+  public void testRenameOverwriteExistingFiles() throws Exception {
+    // Simulate a filesystem which will not overwrite existing files
+    FileSystem<LocalResourceId> fs = new LocalFileSystem(new CopyOption[0]);
+    executeRename(fs, true);
+  }
+
+  /**
+   * Creates two resources renaming them where one of the target files already exists.
+   *
+   * @param fs The internal file system implementation
+   * @param replaceExisting True if an attempt to replace existing files is desired
+   * @throws Exception On error creating the files or renaming
+   */
+  private void executeRename(FileSystem<LocalResourceId> fs, boolean replaceExisting)
```

Comment:
Probably rename to something like executeRenameWithAnExistingDestinationFile to convey the intention of this method properly.